src/types/type_wrappers.rs

```diff
@@ -69,6 +69,47 @@ impl ComplexType {
         // Batch query to NAPI minimizes number of calls
         self.inner_types.clone()
     }
+
+    // Create a new copy of the ComplexType, with first support type set to the provided type
+    #[napi]
+    pub fn remap_list_support_type(&self, new_subtype: &ComplexType) -> ComplexType {
+        ComplexType::one_support(self.base_type, Some(new_subtype.clone()))
+    }
+
+    #[napi]
+    // Create a new copy of the ComplexType, with first and second support type set to the provided types
+    pub fn remap_map_support_type(
+        &self,
+        key_new_subtype: &ComplexType,
+        val_new_subtype: &ComplexType,
+    ) -> ComplexType {
+        ComplexType::two_support(
+            self.base_type,
+            Some(key_new_subtype.clone()),
+            Some(val_new_subtype.clone()),
+        )
+    }
```

Comment:
:question: Do these functions change the element type of the collection, leaving the very collection type untouched? I.e., `Set<SmallInt>` becomes e.g. `Set<Ascii>`, and `Map<Int, Float>` becomes `Map<Double, Blob>`.
:wrench: If so, please mention these examples in the docstrings of those functions.

:recycle: Please use doc comments (`///`) instead of plain comments (`//`) to document functions/fields/structs etc., because such documentation pops up on hover.